src/blog/2025/05/how-to-generate-pdf-reports-using-node-red.md

```diff
@@ -0,0 +1,336 @@
+
+When working with this node, you can use various attributes to customize the content and layout of the PDF, such as text, tables, images, page sizes, margins, headers, footers, and more. Below are the most commonly used attributes:
+
+| **Attribute**     | **Description**                                                                 | **Example**                                              |
+|-------------------|---------------------------------------------------------------------------------|----------------------------------------------------------|
+| `content`         | Defines the content of the PDF (text, tables, images, etc.).                    | `{ "content": "Hello, World!" }`                         |
+| `style`           | Specifies the style for content (font size, font family, etc.).                 | `{ "style": "headerStyle" }`                              |
+| `layout`          | Defines the layout of a table (e.g., 'lightHorizontalLines', 'noBorders').      | `{ "layout": "lightHorizontalLines" }`                    |
+| `pageSize`        | Defines the page size for the PDF.                                              | `{ "pageSize": "A4" }`                                    |
+| `pageMargins`     | Sets the margins for the PDF (left, top, right, bottom).                        | `{ "pageMargins": [40, 60, 40, 60] }`                     |
+| `header`          | Specifies a header for the PDF. Can be a static text or dynamic content.        | `{ "header": "My PDF Report" }`                           |
+| `footer`          | Specifies a footer for the PDF. Can be a static text or dynamic content.        | `{ "footer": "Page {PAGE_NUM} of {PAGE_COUNT}" }`         |
+| `defaultStyle`    | Defines the default style for all content in the PDF.                           | `{ "defaultStyle": { "font": "Helvetica", "fontSize": 12 } }` |
+| `background`      | Adds a background to the page or content area.                                  | `{ "background": { "image": "imageData" } }`             |
+| `width`           | Sets the width of table cells or other elements.                                | `{ "width": 150 }`                                        |
+| `height`          | Sets the height of table cells or other elements.                               | `{ "height": 50 }`                                        |
+| `alignment`       | Specifies the text alignment (left, center, right).                             | `{ "alignment": "center" }`                               |
+| `border`          | Defines the border for tables or table cells (style, width, and color).         | `{ "border": [true, true, true, true] }`                   |
```

Comment:
(optional)
Should we include a link to pdf docs for the underlying technology used here? 
https://pdfmake.github.io/docs/0.1/document-definition-object/styling/